### DISCUSSION TITLE
Allowing key repetition at the time of creation

### DISCUSSION THREAD

**Discussion by AndresDuque** [nlohmann/json#3778] [Q&A] [answered]
Hi there,

I would like to create the following structure with the nlohmann json library:
"images" :[
  {
  "image_path" : "0bfeb4c3-eb5a-446f-8d18-f1cf4995dbbb.PNG",
  "type": "RPE"
  },
  {
  "image_path" : "0fc19b68-0ba0-42ff-b549-c9fa81949aae.PNG",
  "type" : "NFL"
  },
  {
  "image_path" : "1acd8b43-f070-4367-9d1a-5fa3969e4aae.PNG",
  "type" : "PR"
  }

which I can when I declare a json variable.

json object = R"(
{
  "images" :[
  {
  "image_path" : "0bfeb4c3-eb5a-446f-8d18-f1cf4995dbbb.PNG",
  "type": "RPE"
  },
  {
  "image_path" : "0fc19b68-0ba0-42ff-b549-c9fa81949aae.PNG",
  "type" : "NFL"
  },
  {
  "image_path" : "1acd8b43-f070-4367-9d1a-5fa3969e4aae.PNG",
  "type" : "PR"
  }
  ]
}
)"_json;

The problem comes when I am passing the data automatically to an empty JSON object, as the image_path key already exists (same for the type) the other data is erased.

How could I achieve this?

Thank you in advance!

**Comment by falbrechtskirchinger** [comment]
What are you trying to do? Show your code.

"(...) empty JSON object (...) image_path key already exists (...)"

The object is either empty, in which case the `image_path` key doesn't exist, or it isn't.

**Reply by AndresDuque** [reply]
Hi @falbrechtskirchinger 

The code is the following:
for (unsigned i = 0; i < patient->getSession()->getImages().size(); i++) {
      image_key_value.insert(
          {{"image_path",
            patient->getSession()->getImages()[i]->getCloudUUID()}});
      image_key_value.insert(
          {{"type",
            imageTypeToStdString(
                patient->getSession()->getImages()[i]->getType())}});
    }

When I serialize de output I just get one value out of the three., that's because the image_path and type already exists for the second iteration.

**Comment by gregmarr** [comment, marked as answer]
Sounds like you created `images` as an object instead of an array, and you're trying to set the values directly on `images`, when you should be creating new objects and adding them to the `images` array.

**Reply by falbrechtskirchinger** [reply]
Shhh. I'm trying to train him to post relevant code with his questions. ;-)

**Reply by AndresDuque** [reply]
Thanks for the insight! Now I got it working with the following code:

object["images"] = json::array({});
  for (unsigned i = 0; i < patient->getSession()->getImages().size(); i++) {
    json images = json::array();
    images = {
        {"images_path", patient->getSession()->getImages()[i]->getCloudUUID()},
        {"type",
         imageTypeToStdString(
             patient->getSession()->getImages()[i]->getType())}};
    object["images"].push_back(images);
  }

**Reply by falbrechtskirchinger** [reply]
```cpp
json images = json::array();
```

You don't need this line since you're overwriting it in the next.

```cpp
json images({
	{"images_path", patient->getSession()->getImages()[i]->getCloudUUID()},
	{"type", imageTypeToStdString( patient->getSession()->getImages()[i]->getType())}
});
```
